tests/coff.rs

```diff
@@ -0,0 +1,25 @@
+use object::{pe, read, Object, ObjectSection};
+use std::fs;
+use std::path::PathBuf;
+
+#[cfg(feature = "coff")]
+#[test]
+fn coff_extended_relocations() {
+    let path_to_obj: PathBuf = ["testfiles", "coff", "relocs_overflow.obj"]
+        .iter()
+        .collect();
+    let contents = fs::read(&path_to_obj).expect("Could not read relocs_overflow.obj");
+    let file =
+        read::coff::CoffFile::parse(&contents[..]).expect("Could not parse relocs_overflow.obj");
+    let code_section = file
+        .section_by_name(".text")
+        .expect("Could not find .text section in relocs_overflow.obj");
+    match code_section.flags() {
+        object::SectionFlags::Coff { characteristics } => {
+            assert!(characteristics & pe::IMAGE_SCN_LNK_NRELOC_OVFL != 0)
+        }
+        _ => panic!("Invalid section flags flavour."),
+    };
+    let relocations = code_section.relocations().collect::<Vec<_>>();
+    assert_eq!(relocations.len(), 65536);
+}
```

Comment:
The alternative way to test this is to add some readobj and objdump files under `crates/examples/testfiles/coff`. Note that these files can be fragments of the output, so we can we check for only the first and last relocation for example.

We should also fix `src/write/coff.rs` and add a roundtrip test for that, but I'm okay with leaving that for another PR.

Reply:
Re alternative test: That sounds like a better idea.
Re write support for extended relocations: Yeah, I think it'd be better to handle it in a separate PR.

Reply:
I believe `readobj` prints the raw values of section's relocations count, which in this case is 65535. I'll leave the current testing appproach as is and wait for your call. CI should be happy now at the very least.

It's even a bit trickier: `readobj` can print all extended relocations (as it uses `.coff_relocations` under the hood), yet reported value is that from header. See the [object_readobj_impl_example.readobj.txt](https://github.com/gimli-rs/object/files/7566325/object_readobj_impl_example.readobj.txt) - first relocation with count of relocations is not present in output, yet the NumberOfRelocations comes from section header. I assume that the value is taken from header as seen [here](https://github.com/gimli-rs/object/blob/master/crates/examples/src/readobj/pe.rs#L340)

Reply:
Yes, `readobj` does and should print the raw header values. It maybe should also print the first relocation containing the extended count, but that's a bit ugly to do.